subgraphs/erc20/src/mappings/token.ts

```diff
@@ -0,0 +1,413 @@
+import {
+  BigDecimal,
+  Bytes,
+  BigInt,
+  ethereum,
+  log,
+} from "@graphprotocol/graph-ts";
+
+import { ERC20, Transfer } from "../../generated/templates/StandardToken/ERC20";
+import { Burn } from "../../generated/templates/BurnableToken/Burnable";
+import { Mint } from "../../generated/templates/MintableToken/Mintable";
+
+import {
+  Token,
+  TokenDailySnapshot,
+  TokenHourlySnapshot,
+  TransferEvent,
+} from "../../generated/schema";
+
+import {
+  SECONDS_PER_HOUR,
+  SECONDS_PER_DAY,
+  GENESIS_ADDRESS,
+  BIGINT_ZERO,
+  BIGINT_ONE,
+  BIGINT_TWO,
+  BIGDECIMAL_ZERO,
+  DEFAULT_DECIMALS,
+} from "../common/constants";
+
+import { toDecimal } from "./registry";
+import {
+  getOrCreateAccount,
+  getOrCreateAccountBalance,
+  increaseAccountBalance,
+  decreaseAccountBalance,
+  isNewAccount,
+  updateAccountBalanceDailySnapshot,
+} from "./account";
+
+export function handleTransfer(event: Transfer): void {
+  let token = Token.load(event.address.toHex());
+
+  if (token != null) {
+    if (token.name == "") {
+      let erc20 = ERC20.bind(event.address);
+      let tokenName = erc20.try_name();
+      let tokenSymbol = erc20.try_symbol();
+      let initialSupply = erc20.try_totalSupply();
+      let tokenDecimals = erc20.try_decimals();
+
+      token.name = tokenName.reverted ? "" : tokenName.value;
+      token.symbol = tokenSymbol.reverted ? "" : tokenSymbol.value;
+      token.decimals = tokenDecimals.reverted
+        ? DEFAULT_DECIMALS
+        : tokenDecimals.value;
+      token.totalSupply = initialSupply.reverted
+        ? token.totalSupply
+        : toDecimal(initialSupply.value, token.decimals);
+    }
+
+    if (event.params.value == BIGINT_ZERO) {
+      return;
+    }
+    let amount = toDecimal(event.params.value, token.decimals);
+
+    let isBurn = event.params.to.toHex() == GENESIS_ADDRESS;
+    let isMint = event.params.from.toHex() == GENESIS_ADDRESS;
+    let isTransfer = !isBurn && !isMint;
+
+    if (isBurn) {
+      handleBurnEvent(token, amount, event.params.from, event);
+    } else if (isMint) {
+      log.error("isMint", []);
+      handleMintEvent(token, amount, event.params.to, event);
```

Comment:
If a mint/burn event is emitted with a transfer event, won't the `handleBurnEvent`/`handleMintEvent` functions run twice?  shouldn't we add a condition for this?

Reply:
Yes. In some library a special transfer function  emit both transfer and mint/burn event. I checked whether on chain totalSupply has been updated to decides the event related with the transfer call has been processed before in today's commit https://github.com/messari/subgraphs/pull/546/commits/c877e3aa03afd264dbb14be0131529662edbc13c Thanks!